django_ipam/base/admin.py

```diff
@@ -47,11 +53,46 @@ def change_view(self, request, object_id, form_url="", extra_context=None):
             extra_context = {'labels': labels,
                              'values': values,
                              'used_ip': used_ip,
+                             'subnet': instance,
                              'ipaddress_add_url': ipaddress_add_url,
                              'ipaddress_change_url': ipaddress_change_url}
 
         return super(AbstractSubnetAdmin, self).change_view(request, object_id, form_url, extra_context)
 
+    def get_urls(self):
+        urls = super().get_urls()
+        custom_urls = [
+            re_path(r'^(?P<subnet_id>[^/]+)/export-ipaddress/',
+                    self.export_ipaddress,
+                    name="export_ipaddress"),
+            path('import/', self.import_ipaddress, name="import_ipaddress"),
+        ]
+        return custom_urls + urls
+
+    def export_ipaddress(self, request, subnet_id):
+        response = HttpResponse(content_type='text/csv')
+        response['Content-Disposition'] = 'attachment; filename="ip_address.csv"'
+        writer = csv.writer(response)
+        Subnet.export_csv(self, subnet_id, writer)
+        return response
+
+    def import_ipaddress(self, request):
+        form = IpAddressImportForm()
+        form_template = 'admin/django-ipam/subnet/import.html'
+        if request.method == 'POST':
+            form = IpAddressImportForm(request.POST, request.FILES)
+            if form.is_valid():
+                file = request.FILES['csvfile']
+                if not file.name.endswith(('.csv', '.xls', '.xlsx')):
+                    messages.error(request, 'File type not supported.')
+                    return render(request, form_template, {'form': form})
+                if not Subnet.import_csv(self, file):
+                    messages.error(request, 'Import data error.')
```

Comment:
here you are not informing the user about what caused the error. This is not good because users need to know what they did wrong in order to fix it and complete the import process.
You should catch an exception here and get the details from the exception. I will explain below.

Keep in mind the` i18n` issue explained in the comment above.